api/configuration/v1alpha1/credential_basic_auth_types.go

```diff
@@ -0,0 +1,159 @@
+
+	// Spec contains the BasicAuth credential specification.
+	Spec CredentialBasicAuthSpec `json:"spec"`
+
+	// Status contains the BasicAuth credential status.
+	//
+	// +kubebuilder:default={conditions: {{type: "Programmed", status: "Unknown", reason:"Pending", message:"Waiting for controller", lastTransitionTime: "1970-01-01T00:00:00Z"}}}
+	Status CredentialBasicAuthStatus `json:"status,omitempty"`
+}
+
+func (s *CredentialBasicAuth) initKonnectStatus() {
+	s.Status.Konnect = &konnectv1alpha1.KonnectEntityStatusWithControlPlaneRef{}
+}
+
+// GetKonnectStatus returns the Konnect status contained in the CredentialBasicAuth status.
+func (s *CredentialBasicAuth) GetKonnectStatus() *konnectv1alpha1.KonnectEntityStatus {
+	if s.Status.Konnect == nil {
+		return nil
+	}
+	return &s.Status.Konnect.KonnectEntityStatus
+}
+
+// GetKonnectID returns the Konnect ID in the CredentialBasicAuth status.
+func (s *CredentialBasicAuth) GetKonnectID() string {
+	if s.Status.Konnect == nil {
+		return ""
+	}
+	return s.Status.Konnect.ID
+}
+
+// SetKonnectID sets the Konnect ID in the CredentialBasicAuth status.
+func (s *CredentialBasicAuth) SetKonnectID(id string) {
+	if s.Status.Konnect == nil {
+		s.initKonnectStatus()
+	}
+	s.Status.Konnect.ID = id
+}
+
+// GetControlPlaneID returns the ControlPlane ID in the CredentialBasicAuth status.
+func (s *CredentialBasicAuth) GetControlPlaneID() string {
+	if s.Status.Konnect == nil {
+		return ""
+	}
+	return s.Status.Konnect.ControlPlaneID
+}
+
+// SetControlPlaneID sets the ControlPlane ID in the CredentialBasicAuth status.
+func (s *CredentialBasicAuth) SetControlPlaneID(id string) {
+	if s.Status.Konnect == nil {
+		s.initKonnectStatus()
+	}
+	s.Status.Konnect.ControlPlaneID = id
+}
+
+// GetTypeName returns the CredentialBasicAuth Kind name
+func (s CredentialBasicAuth) GetTypeName() string {
+	return "CredentialBasicAuth"
+}
+
+// GetConditions returns the Status Conditions
+func (s *CredentialBasicAuth) GetConditions() []metav1.Condition {
+	return s.Status.Conditions
+}
+
+// SetConditions sets the Status Conditions
+func (s *CredentialBasicAuth) SetConditions(conditions []metav1.Condition) {
+	s.Status.Conditions = conditions
+}
+
+// CredentialBasicAuthSpec defines specification of a Kong Route.
+type CredentialBasicAuthSpec struct {
+	// ConsumerRef is a reference to a Consumer this CredentialBasicAuth is associated with.
+	//
+	// +kubebuilder:validation:Required
+	ConsumerRef corev1.LocalObjectReference `json:"consumerRef"`
+
+	// SecretRef is a reference to a Secret this CredentialBasicAuth is associated with.
+	//
+	// +kubebuilder:validation:Required
+	SecretRef corev1.LocalObjectReference `json:"secretRef"`
+
+	CredentialBasicAuthAPISpec `json:",inline"`
+}
+
+// CredentialBasicAuthAPISpec defines specification of a BasicAuth credential.
+type CredentialBasicAuthAPISpec struct {
+	// Password is the password for the BasicAuth credential.
+	Password *string `json:"password,omitempty"`
```

Comment:
Should it be redacted?

Reply:
Hm, that's an interesting idea. Looking at https://book.kubebuilder.io/reference/markers I cannot find a code marker that would allow this. Quick google search doesn't yield any results about this feature of CRD fields. 🤔 

---

An alternative is to not copy the config from the Secret but to get the config inside the generic reconciler when Konnect entity is to be created/updated. I'd prefer to avoid that as that would make the generic reconciler be responsible for even more, whereas with the approach from this PR it will not need to apply what's defined in the CRD. 

Plus, with this approach users can clearly see in the CRD what's the expected state in Konnect.